src/Http/Middleware/CspMiddleware.php

```diff
@@ -0,0 +1,87 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         4.0.0
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+
+namespace Cake\Http\Middleware;
+
+use Cake\Core\Configure;
+use InvalidArgumentException;
+use ParagonIE\CSPBuilder\CSPBuilder;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+
+/**
+ * Content Security Policy Middleware
+ */
+class CspMiddleware implements MiddlewareInterface
+{
+    /**
+     * CSP Builder
+     *
+     * @var \ParagonIE\CSPBuilder\CSPBuilder $csp CSP Builder or config array
+     */
+    protected $csp;
+
+    /**
+     * Constructor
+     *
+     * @param \ParagonIE\CSPBuilder\CSPBuilder|array|null $csp CSP object or config array
+     * @throws \Exception
+     */
+    public function __construct($csp = null)
+    {
+        if ($csp === null) {
+            $cspConfig = (array)Configure::read('App.CSP');
+        } else {
+            $cspConfig = $csp;
+        }
+
+        if (!empty($cspConfig) && is_array($cspConfig)) {
+            $this->csp = CSPBuilder::fromData(json_encode($cspConfig));
```

Comment:
Couldn't the config be passed directly to `CSPBuilder::fromArray()`, or the `CSPBuilder` constructor?

Reply:
It now also possible to pass a `CSPBuilder` instance to `CspMiddleware` see https://github.com/cakephp/cakephp/pull/13252/files#diff-3faf87dd86f33d27dcb8b089b5644176R96

Reply:
@landenh But why json encode the array and pass to fromData() instead of using fromArray()?

Reply:
ohh, sorry I misunderstood